.chloggen/case-sensitive-configuration.yaml

```diff
@@ -0,0 +1,16 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: 'bug_fix'
```

Comment:
Is this a breaking change? Users who have the wrong case will now receive an error when starting their collector.

Reply:
possible. Is there any implication besides the changelog? 

I'm a bit resistant on changing though because essentially every bug is a breaking change. Users might rely on buggy behavior and when you fix it you can cause a regression.

Reply:
There's always a [relevant XKCD](https://xkcd.com/1172/).

I don't think this is a breaking change.  It will cause configurations that are already failing to fail loudly and quickly.